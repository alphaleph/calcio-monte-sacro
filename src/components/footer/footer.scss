@import '../../pages/globals.scss';

.footer-container {
    width: 100%;
    font-size: 1.25rem;
    text-align: center;
    margin-top: 0rem !important;
    margin-left: 0rem !important;
    margin-right: 0rem !important;
}

.footer-divider {
    width: 70%;
    height: 1px;
    background: $holy-yellow-team;
    position: relative;
    left: 15%;
}

.footer-main-links {
    padding-top: 3rem;
    padding-bottom: 3rem;
}

.footer-main-link {
    margin: 0rem 1rem;
    color: $white-main;
    white-space: pre;
}

.footer-main-link:hover,
.footer-main-link:focus {
    text-decoration: underline;
    color: $holy-yellow-team;
}

.footer-legal-buttons {
    padding-top: 3rem;
    padding-bottom: 3rem;
}

.footer-legal-button {
    background: none;
    border: none;
    padding: 0;
    color: $white-main;
    font-size: 1.25rem;
    margin: 0rem 1rem;
}

.footer-legal-button:visited,
.footer-legal-button:hover,
.footer-legal-button:focus {
    text-decoration: underline;
    color: $holy-yellow-team;
}

.copyright {
    color: $granite-grey-team-invert;
    font-size: 1rem;
    padding-bottom: 1rem;
}

.copyright .email-link {
    margin-left:.5rem;
    color: $granite-grey-team-invert;
}

.copyright .email-link:visited,
.copyright .email-link:focus,
.copyright .email-link:hover {
    text-decoration: underline;
    color: $roma-tyrian-invert;
}

@media screen and (max-width: 1023px) {
    .footer-divider {
        width: 60%;
        left: 20%;
    }
}