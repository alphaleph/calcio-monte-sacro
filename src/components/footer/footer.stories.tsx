import React from 'react';
import { storiesOf } from '@storybook/react';
import { Footer } from './footer';
import '../../pages/index.scss';

storiesOf('Footer', module).add('default', () => <Footer />);
