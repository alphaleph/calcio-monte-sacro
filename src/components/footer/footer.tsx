import React from 'react';
import { Link } from 'gatsby';
import './footer.scss';
import {
    SITE_URL,
    EMAIL_ADDR,
    TERMS_AND_CONDITIONS,
    PRIVACY,
    COOKIES,
} from '../../constants';

interface FooterProps {
    openModal: (modalType: string) => void;
}

export const Footer = (props: FooterProps) => {
    const openTermsAndCondModal = () => {
        props.openModal(TERMS_AND_CONDITIONS);
    };

    const openPrivacyModal = () => {
        props.openModal(PRIVACY);
    };

    const openCookiesModal = () => {
        props.openModal(COOKIES);
    };

    return (
        <footer className="footer-container columns is-centered has-background-team-granite">
            <div className="column is-two-thirds">
                <section className="footer-main-links container">
                    <Link
                        className="footer-main-link"
                        to={`${SITE_URL}/schedule`}
                    >
                        Schedule
                    </Link>
                    <Link className="footer-main-link" to={`${SITE_URL}/news`}>
                        News
                    </Link>
                    <Link className="footer-main-link" to={`${SITE_URL}/blog`}>
                        Blog
                    </Link>
                    <Link
                        className="footer-main-link"
                        to={`${SITE_URL}/drills`}
                    >
                        Drills
                    </Link>
                    <Link className="footer-main-link" to={`${SITE_URL}/about`}>
                        About
                    </Link>
                </section>
                <div className="footer-divider" />
                <section className="footer-legal-buttons container">
                    <button
                        type="button"
                        className="footer-legal-button"
                        onClick={openTermsAndCondModal}
                    >
                        Terms and Conditions
                    </button>
                    <button
                        type="button"
                        className="footer-legal-button"
                        onClick={openPrivacyModal}
                    >
                        Privacy
                    </button>
                    <button
                        type="button"
                        className="footer-legal-button"
                        onClick={openCookiesModal}
                    >
                        Cookies
                    </button>
                </section>
                <section className="copyright container">
                    <p>
                        Â© 2019 Calcio Monte Sacro. All Rights Reserved. Website
                        questions, comments or suggestions?
                        <a className="email-link" href={`mailto:${EMAIL_ADDR}`}>
                            Email Us
                        </a>
                    </p>
                </section>
            </div>
        </footer>
    );
};
