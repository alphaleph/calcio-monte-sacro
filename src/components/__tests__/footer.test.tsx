import React from 'react';
import { Footer } from '../footer/footer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import { resizeToMobile, mainLinkTexts, legalButtonTexts } from './test-util';
import {
    EMAIL_ADDR,
    TERMS_AND_CONDITIONS,
    PRIVACY,
    COOKIES,
} from '../../constants';

const openModal = jest.fn();

describe('Footer', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('renders without crashing', () => {
        const { getByText } = render(<Footer openModal={openModal} />);
        expect(getByText(EMAIL_ADDR)).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<Footer openModal={openModal} />);
        for (let pageTitle of Object.keys(mainLinkTexts)) {
            expect(getByText(pageTitle)).toBeVisible();
            expect(getByText(pageTitle)).toHaveAttribute(
                'href',
                mainLinkTexts[pageTitle]
            );
        }
        for (let buttonText of legalButtonTexts) {
            expect(getByText(buttonText)).toBeVisible();
        }
        expect(getByText(EMAIL_ADDR)).toBeVisible();
    });

    it('renders correctly on mobile', () => {
        resizeToMobile();
        const { getByText } = render(<Footer openModal={openModal} />);

        for (let pageTitle of Object.keys(mainLinkTexts)) {
            expect(getByText(pageTitle)).toBeVisible();
            expect(getByText(pageTitle)).toHaveAttribute(
                'href',
                mainLinkTexts[pageTitle]
            );
        }
        for (let buttonText of legalButtonTexts) {
            expect(getByText(buttonText)).toBeVisible();
        }
        expect(getByText(EMAIL_ADDR)).toBeVisible();
    });

    it('opens Terms and Conditions modal on correct button click', () => {
        const { getByText } = render(<Footer openModal={openModal} />);
        userEvent.click(getByText(legalButtonTexts[0]));
        expect(openModal).toHaveBeenCalledTimes(1);
        expect(openModal).toHaveBeenCalledWith(TERMS_AND_CONDITIONS);
    });

    it('opens Privacy modal on correct button click', () => {
        const { getByText } = render(<Footer openModal={openModal} />);
        userEvent.click(getByText(legalButtonTexts[1]));
        expect(openModal).toHaveBeenCalledTimes(1);
        expect(openModal).toHaveBeenCalledWith(PRIVACY);
    });

    it('opens Cookies modal on correct button click', () => {
        const { getByText } = render(<Footer openModal={openModal} />);
        userEvent.click(getByText(legalButtonTexts[2]));
        expect(openModal).toHaveBeenCalledTimes(1);
        expect(openModal).toHaveBeenCalledWith(COOKIES);
    });

    it('has not changed', () => {
        const { asFragment } = render(<Footer openModal={openModal} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
