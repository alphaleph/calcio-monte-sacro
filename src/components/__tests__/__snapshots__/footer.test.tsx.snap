// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer has not changed 1`] = `
<DocumentFragment>
  <footer
    class="footer-container columns is-centered has-background-team-granite"
  >
    <div
      class="column is-two-thirds"
    >
      <section
        class="footer-main-links container"
      >
        <a
          class="footer-main-link"
          href="https://www.calciomontesacro.com/schedule"
        >
          Schedule
        </a>
        <a
          class="footer-main-link"
          href="https://www.calciomontesacro.com/news"
        >
          News
        </a>
        <a
          class="footer-main-link"
          href="https://www.calciomontesacro.com/blog"
        >
          Blog
        </a>
        <a
          class="footer-main-link"
          href="https://www.calciomontesacro.com/drills"
        >
          Drills
        </a>
        <a
          class="footer-main-link"
          href="https://www.calciomontesacro.com/about"
        >
          About
        </a>
      </section>
      <div
        class="footer-divider"
      />
      <section
        class="footer-legal-buttons container"
      >
        <button
          class="footer-legal-button"
          type="button"
        >
          Terms and Conditions
        </button>
        <button
          class="footer-legal-button"
          type="button"
        >
          Privacy
        </button>
        <button
          class="footer-legal-button"
          type="button"
        >
          Cookies
        </button>
      </section>
      <section
        class="copyright container"
      >
        <p>
          Â© 2019 Calcio Monte Sacro. All Rights Reserved. Website questions, comments or suggestions?
          <a
            class="email-link"
            href="mailto:&#099;&#097;&#108;&#099;&#105;&#111;&#109;&#111;&#110;&#116;&#101;&#115;&#097;&#099;&#114;&#111;&#064;&#103;&#109;&#097;&#105;&#108;&#046;&#099;&#111;&#109"
          >
            Email Us
          </a>
        </p>
      </section>
    </div>
  </footer>
</DocumentFragment>
`;
